src/providers/postgres/qgspostgresdataitemguiprovider.cpp

```diff
@@ -735,3 +760,239 @@ void QgsPostgresDataItemGuiProvider::handleImportVector( QgsPGConnectionItem *co
+    conn->unref();
+    return;
+  }
+
+  // refresh
+  projectItem->parent()->refresh();
+
+  conn->unref();
+}
+
+void QgsPostgresDataItemGuiProvider::deleteProject( QgsPGProjectItem *projectItem, QgsDataItemGuiContext context )
+{
+  if ( QMessageBox::question( nullptr, tr( "Delete Project" ), tr( "Are you sure you want to delete project “%1”?" ).arg( projectItem->name() ), QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
+    return;
+
+  QgsPostgresConn *conn = QgsPostgresConn::connectDb( projectItem->postgresProjectUri().connInfo, false );
+  if ( !conn )
+  {
+    notify( tr( "Delete Project" ), tr( "Unable to delete project." ), context, Qgis::MessageLevel::Warning );
+    return;
+  }
+
+  if ( !QgsPostgresUtils::deleteProjectFromSchema( conn, projectItem->name(), projectItem->schemaName() ) )
+  {
+    notify( tr( "Delete Project" ), tr( "Unable to delete project “%1” " ).arg( projectItem->name() ), context, Qgis::MessageLevel::Warning );
+    conn->unref();
+    return;
+  }
+
+  // refresh
+  projectItem->parent()->refresh();
+
+  conn->unref();
+}
+
+void QgsPostgresDataItemGuiProvider::duplicateProject( QgsPGProjectItem *projectItem, QgsDataItemGuiContext context )
+{
+  QgsNewNameDialog dlg( tr( "Project “%1”" ).arg( projectItem->name() ) );
+  dlg.setWindowTitle( tr( "Duplicate Project" ) );
+  if ( dlg.exec() != QDialog::Accepted || dlg.name() == projectItem->name() )
+    return;
+
+  QgsPostgresConn *conn = QgsPostgresConn::connectDb( projectItem->postgresProjectUri().connInfo, false );
+  if ( !conn )
+  {
+    notify( tr( "Duplicate Project" ), tr( "Unable to duplicate project." ), context, Qgis::MessageLevel::Warning );
+    return;
+  }
+
+  QString newUri = projectItem->uriWithNewName( dlg.name() );
+
+  // read the project, set title and new filename
+  QgsProject project = QgsProject();
+  project.read( projectItem->path() );
+  project.setTitle( dlg.name() );
+  project.setFileName( newUri );
+
+  // write project to the database
+  bool success = project.write();
+  if ( !success )
+  {
+    notify( tr( "Duplicate Project" ), tr( "Unable to duplicate project “%1” to “%2”" ).arg( projectItem->name(), dlg.name() ), context, Qgis::MessageLevel::Warning );
+    conn->unref();
+    return;
+  }
+
+  // refresh
+  projectItem->parent()->refresh();
+}
+
+void QgsPostgresDataItemGuiProvider::moveProjectToSchema( QgsPGProjectItem *projectItem, QgsDataItemGuiContext context )
+{
+  QgsPostgresSchemaSelectionDialog *dlg = new QgsPostgresSchemaSelectionDialog( projectItem->postgresProjectUri().connInfo );
+
+  if ( dlg->exec() == QDialog::Accepted )
+  {
+    QgsPostgresConn *conn = QgsPostgresConn::connectDb( projectItem->postgresProjectUri().connInfo, false );
+    if ( !conn )
+    {
+      notify( tr( "Move Project to Another Schema" ), tr( "Unable to move project to anoter schema." ), context, Qgis::MessageLevel::Warning );
+      return;
+    }
+
+    QString newSchemaName = dlg->schema();
```

Comment:
```suggestion
    const QString newSchemaName = dlg->schema();
```

there are probably other `const` missing but I'll leave it to you.